apps/prairielearn/elements/pl-order-blocks/dag_checker.py

```diff
@@ -55,6 +62,25 @@ def solve_dag(
     return sort
 
 
+def solve_multigraph(
+    depends_multi_graph: Multigraph,
+    final: str,
+) -> list[list[str]]:
+    """Solve the given problem
+    :param depends_multi_graph: the dependency multi graph specified in the question
+    :param final: the sink of the multigraph
+    :return: a list of lists that are a topological sort of the input MDAG making it a solution to the given problem
+    """
+    graphs = [
+        dag_to_nx(graph, {})
+        for graph in collapse_multigraph(depends_multi_graph, final)
+    ]
+
+    sort = [list(nx.topological_sort(graph)) for graph in graphs]
+
+    return sort
```

Comment:
nit: the whitespace here is a little bit overkill because its a short function, I don't think you need any blank lines for spacing